src/core_plugins/kibana/public/discover/directives/no_results.js

```diff
@@ -1,12 +1,195 @@
-import { uiModules } from 'ui/modules';
-import noResultsTemplate from '../partials/no_results.html';
-import 'ui/directives/documentation_href';
-
-uiModules
-  .get('apps/discover')
-  .directive('discoverNoResults', function () {
-    return {
-      restrict: 'E',
-      template: noResultsTemplate
+import React, { Component, Fragment } from 'react';
+import PropTypes from 'prop-types';
+
+import {
+  EuiCallOut,
+  EuiCode,
+  EuiCodeBlock,
+  EuiDescriptionList,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiLink,
+  EuiSpacer,
+  EuiText,
+} from '@elastic/eui';
+
+export class DiscoverNoResults extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isTimePickerOpen: false,
```

Comment:
This assumes the timePicker is not open when the `DiscoverNoResults` component is rendered, right?

Perhaps `topNavToggle` should become two props: `toggleTimePicker` and `isTimePickerOpen` so that the topNav controller can inform the `DiscoverNoResults`?